composer.json

```
@@ -11,7 +11,7 @@
         }
     ],
     "require": {
-        "php": ">=5.3.0"
+        "php": ">=7.2.0"
```

Please do not use `>=` (or, worse, `*`) for constraints; there's no way of knowing if a later major will result in BC breaks.

```suggestion
        "php": "^7.2 || ^8.0"
```

---

We don't have any bylaws for it, and many of these pre-dated some of our current understanding of what is reasonable to expect for backwards compatibility guarantees from PHP.

Laminas, for instance, uses `~X.Y.0` constraints, because we've found that there are subtle BC breaks often between minor versions.

Considering we do not know what keywords will be reserved for a new major, nor what subtle changes to the object model might cause breakage, I feel we should be cautious in what we say we support. We can always create a new minor to add support for a new PHP major when the time comes, but _removing_ support for the new major if we discover breakage would technically be a BC break.

---

I think we both agree this would not be the right place to have this discussion and I think we're both very well aware of the pros and cons of both approaches. Neither approach is perfect, it's a tradeoff. Hence my question if there is (or should be) a bylaw that goes into more detail.

In my opinion, it really comes down to expectations and how we (can) look into our crystal ball. I would argue I have no reason to believe a set of interfaces like PSR-7 would be incompatible with PHP 9 and beyond – or at least I have equal certainty it will be compatible with PHP 8.4 or beyond. Both assumptions are guess work at this point imho, so I could even understand the move to pin to the minor version like done for Laminas and others.

I guess I could rephrase my thoughts: Do we have more reason to believe PHP 9 would break PSR-7 v2 than we have reason to believe that a future PSR-7 v2.x would be ready in time PHP 9 is foreseeable? I would hate to hinder adoption of new PHP versions in the future.